platform-sdk/docs/proposals/consensus-layer/Consensus-Layer.md

```diff
@@ -0,0 +1,709 @@
+Execution may look at each transaction within the self-event, and decide that some transactions (such as those that have
+expired or will soon expire) should be dropped while others (such as those not close to expiration) should be
+resubmitted in the next event.
+
+### Sheriff Module
+
+When misbehavior is found for a node, it is reported to the Sheriff. This module keeps track of the different types of
+misbehavior each node is accused of, and uses this information to determine whether to "shun" or "welcome" a node. It
+also sends this information to the Execution layer, so it may record misbehavior in state, if it so chooses, or publish
+misbehavior to other nodes in the network, allowing the network as a whole to observe and report dishonest or broken
+nodes.
+
+![Sheriff](sheriff-module.png)
+
+A node may "shun" another node, by refusing to talk with it via Gossip. If a node were to be shunned by all its
+gossip neighbors, then it has been effectively removed from the network, as it can no longer submit events that will be
+spread through the network, and will therefore not contribute to consensus. Should a malicious node attempt to attack
+its neighbors, if those neighbors discover this attack and simply shun the node, then taking no other action, the
+malicious node is prevented from causing further harm to the network.
+
+It may be that a node is misbehaving due to a bug, or environmental issue, rather than due to malicious intent. For
+example, a broken node with network trouble may attempt to create many connections, as each prior connection having
+failed for some reason. But it could also be malicious intent. Unable to tell the difference, the Sheriff may decide to
+shun the node for some time period, and then "welcome" it back by allowing it to form connections again. It is up to
+the Sheriff's algorithms to decide on the correct response to different behaviors. These algorithms are not defined
+here, but will be defined within the Sheriff's design documentation.
+
+### Public API
+
+![Consensus API](consensus-api.png)
+
+The public API of the Consensus module forms the interface boundary between Consensus and Execution.
+
+#### initialize
+
+Called by the Execution layer when it creates a new Consensus instance. This call will include any required arguments
+to fully initialized Consensus. For example, it will include any initial state corresponding to the current round
+at which Consensus should begin work (such as the correct set of judges for that particular round). Each module within
+Consensus supports some kind of initialization API that will be called. For example, each module must be initialized
+with the initial roster for the starting round.
+
+#### destroy
+
+Called by the Execution layer when it destroys an existing Consensus instance. This is particularly needed to shut down
+network connections held by Gossip, but could be used to stop executors, background tasks, etc. as necessary.
+
+#### changeRoster
```

Comment:
Is this the same method that is used to implement `Consensus only produces rounds on demand. Execution will ask Consensus to produce the next round.` ? If so, then its name is a bit misleading. If not, then I don't see an API for requesting the next round.

Reply:
I think you are right @anthony-swirldslabs, it seems this is the method the instructs Consensus to produce another round, and Execution can optionally provide a new roster. If that is true, I think something like `nextRound` is more intuitive.

Reply:
I also prefer `nextRound`.

@rbair23 Could you weigh in here, since it's a meaningful modification of the API?

Reply:
Agreed 👍🏻